src/windows/main_window.py

```diff
@@ -1743,16 +1743,15 @@ def actionRemove_from_Project_trigger(self):
             if not f:
                 continue
 
-            f_id = f.data["id"]
-            # Remove file
-            f.delete()
-
             # Find matching clips (if any)
-            clips = Clip.filter(file_id=f_id)
+            clips = Clip.filter(file_id=f.data.get("id"))
             for c in clips:
                 # Remove clip
                 c.delete()
 
+            # Remove file (after clips are deleted)
```

Comment:
Moving File deletion AFTER clip deletion (to allow the undo action to process things in the correct order, to prevent missing thumbnails)